release/models/ospf/openconfig-ospfv2-area-interface.yang

```diff
@@ -227,6 +233,58 @@ submodule openconfig-ospfv2-area-interface {
     }
   }
 
+  grouping ospfv2-area-interface-authentication-config {
+    description
+      "Configuration parameters relating to OSPF interface
+      authentication";
+
+    leaf auth-mode {
+      type identityref {
+        base oc-ospf-types:AUTH_MODE;
+      }
+      description
+        "Authentication mode to be used for OSPF interface links.";
+    }
+
+    leaf simple-password {
+      type string;
+      description
+        "Password for simple authentication mode to be specified for OSPF
+        authentication.";
+    }
+  }
+
+  grouping ospfv2-area-interface-message-digest-config {
+    description
+      "Configuration parameters relating to OSPF message-digest
+      authentication";
+
+    leaf key-id {
+      type uint8 {
+        range "1..255";
+      }
+      description
+        "Message-digest keyid to identify auth key and auth algo for
+        OSPF authentication.";
+    }
+
+    leaf auth-algo {
+      type identityref {
+        base oc-ospf-types:MESSAGE_DIGEST_ALGO;
+      }
+      description
+        "Hash algorithm to be used generate/verify message digest
+        for OSPF authentication.";
+    }
+
+    leaf auth-key {
+      type string;
```

Comment:
For this and `simple-password`, we'll want to consider what these values mean for storage.  Much like in ISIS, we likely at least want to leverage `oc-types:routing-password` to put some rules around a specific typedef when it comes to reads and writes.

Reply:
I have changed the simple-password and auth-key type to oc-types:routing-password